### PR TITLE
new(modern_bpf): add support for semget and semctl syscall

### PR DESCRIPTION
**Pull request by loresuso**
Signed-off-by: Lorenzo Susini <susinilorenzo1@gmail.com>

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area API-version

> /area build

> /area CI

> /area driver-kmod

> /area driver-bpf

/area driver-modern-bpf

> /area libscap-engine-bpf

> /area libscap-engine-gvisor

> /area libscap-engine-kmod

> /area libscap-engine-modern-bpf

> /area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-savefile

> /area libscap-engine-udig

> /area libscap

> /area libpman

> /area libsinsp

> /area tests

> /area proposals

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

> /version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

**What this PR does / why we need it**:

Adding support for `semget` and `semctl` in modern bpf. 

**Which issue(s) this PR fixes**:

Part of https://github.com/falcosecurity/libs/issues/723

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: driver/modern_bpf/definitions/events_dimensions.h [hunk 1]
```diff
@@ -160,6 +160,8 @@
 #define IO_URING_SETUP_X_SIZE HEADER_LEN + sizeof(int64_t) + sizeof(uint32_t) * 7 + PARAM_LEN * 8
 #define MMAP2_E_SIZE HEADER_LEN + sizeof(uint64_t) * 3 + sizeof(int64_t) + sizeof(uint32_t) * 2 + PARAM_LEN * 6
 #define MMAP2_X_SIZE HEADER_LEN + sizeof(int64_t) + sizeof(uint32_t) * 3 + PARAM_LEN * 4
+#define SEMGET_E_SIZE HEADER_LEN + sizeof(int32_t) * 2 + sizeof(uint32_t) + PARAM_LEN * 3
+#define SEMGET_X_SIZE HEADER_LEN + sizeof(int64_t) + PARAM_LEN
```
Comment (loresuso) [inline driver/modern_bpf/definitions/events_dimensions.h:164]: Just wanted to notice this: I would put the PARAM_LEN before each sizeof, so to better reflect the layout of the event in memory (header, lengths, actual params). Just a nit-pick, nothing important

File: driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/semget.bpf.c [hunk 2]
```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2022 The Falco Authors.
```
Comment (hbrueckner) [inline driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/semget.bpf.c:2]: nit: 2023

**Review by hbrueckner — COMMENTED** [review]
Thanks @loresuso,

This looks good to me. So feel free to add `semctl` as well.

**Comment by loresuso** [comment]
Thanks @hbrueckner, I've added `semctl` syscall as well

**Comment by FedeDP** [comment]
/milestone next-driver

Great job @loresuso!

**Review by FedeDP — COMMENTED** [review]
This LGTM! Waiting on Andrea/Hendrik feedback ;)

File: driver/modern_bpf/definitions/events_dimensions.h (hunk 1 above)
Comment (Andreagit97) [reply driver/modern_bpf/definitions/events_dimensions.h:164]: agree with that, one day we could also move these sizes to the event table or even better generate them at compile time

**Review by Andreagit97 — APPROVED** [review]
@loresuso thank you for that! 
/approve

File: driver/modern_bpf/definitions/events_dimensions.h (hunk 1 above)
Comment (loresuso) [reply driver/modern_bpf/definitions/events_dimensions.h:164]: Yep exactly, I was thinking about how to generate them too :)

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**